From 8d988c7db129c3fc2d8c8df6f90e2c9e3cd3cad3 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 26 Nov 2019 18:44:20 +0100
Subject: [PATCH 1027/1239] spi: kirkwood: prevent limiting speed to 0

After commit 1fe929ed497bcc8975be8d37383ebafd22b99dd2
("spi: kirkwood: prevent configuring speed exceeding max controller freq")
the spi frequency could be set to 0 on platform where spi-max-frequency
is not defined (e.g. on armada-388-gp). Prevent limiting speed in
mentioned cases.

Change-Id: Ib8d5638b71122f18e345e39fad10384c143050de
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/19345
Tested-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Marcin Wojtas <marcin@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
---
 drivers/spi/kirkwood_spi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/kirkwood_spi.c b/drivers/spi/kirkwood_spi.c
index 8d151cc171..55feda274f 100644
--- a/drivers/spi/kirkwood_spi.c
+++ b/drivers/spi/kirkwood_spi.c
@@ -271,7 +271,7 @@ static int mvebu_spi_set_speed(struct udevice *bus, uint hz)
 	u32 data, divider;
 	unsigned int spr, sppr;
 
-	if (hz > spi->max_hz) {
+	if (spi->max_hz && (hz > spi->max_hz)) {
 		debug("%s: limit speed to the max_hz of the bus %d\n",
 		      __func__, spi->max_hz);
 		hz = spi->max_hz;
-- 
2.29.0

